account-kit/java/signer-sdk/src/main/java/com/alchemy/aa/Stamper.java

```diff
@@ -144,94 +69,35 @@ public void injectCredentialBundle(String bundle) throws GeneralSecurityExceptio
      * @return signed stamp
      *
      * @throws GeneralSecurityException
-     *             iif the private is malformed
+     *             if the private is malformed
      */
-    public Stamp stamp(String payload) throws GeneralSecurityException {
-        if (this.bundlePrivateKey == null || this.bundlePublicKey == null) {
+    public synchronized Stamp stamp(String payload) throws GeneralSecurityException, JsonProcessingException {
```

Comment:
Why is this synchronized? If there's some portion of it that needs synchronization, I'd suggest putting a synchronized block around just that section. It's best to hold locks for as short a time as possible.

Reply:
done